reference/enterprise-api-ref/build-api.markdown

```diff
@@ -0,0 +1,286 @@
+---
+layout: default
+title: Build API
+published: true
+tags: [reference, enterprise, API, build, modules]
+---
+
+The Build API enables you to configure modules easily to your projects.
+
+# Projects API
+
+## Create project
+
+**URI:** https://hub.cfengine.com/build/projects
+
+**Method:** POST
+
+**Parameters:**
+* **repositoryUrl** *(string)*
+  Git repository URL. Project will be synchronized with this repository. Supported protocols: `http`, `ssh`, `git`. Required. 
+
+* **branch** *(string)*
+  Repository branch. Required.
+
+* **name** *(string)*
+  Project name.
+
+* **authenticationType** *(string)*
+  Authentication type that will be used to get access to the repository. Allowed values: `password`, `ssh_key`. Required.
+
+* **username** *(string)*
+  Username for authentication to the repository. Required when authentication type is `password`.
+
+* **password** *(string)*
+  Password for authentication to the repository. Required when authentication type is `password`.
+
+* **sshPrivateKey** *(string)*
+  SSH private key for authentication to the repository.
+
+**Example request (curl):**
+
+```console
+curl -k --user <username>:<password> \
+  -X POST \
+  https://hub.cfengine.com/api/build/projects \
+  -H 'content-type: application/json' \
+  -d '
+  {
+  "repositoryUrl": "https://github.com/username/repository.git",
+  "branch": "master",
+  "authenticationType": "password",
+  "password" : "git_token_or_password",
+  "username" : "git_username",
+  "name": "Production"
+  }'
+```
+
+**Successful response example:**
+
+```
+HTTP 200 Ok
+{
+    "id": "8"
+}
+```
+
+**Responses:**
+
+| HTTP response code | Description |
+|--------|-------------|
+| 200 OK | Project successfully created |
+| 422 Unprocessable entity | Validation error occurred |
+| 409 Conflict | Project with the same repository url and branch already exists |
+| 500 Internal server error | Internal server error |
+
+
+## Update project
+
+**URI:** https://hub.cfengine.com/build/projects/:id
+
+**Method:** PATCH
+
+**Parameters:**
+* **id** *(integer)*
+  Project's ID
+
+* **repositoryUrl** *(string)*
+  Git repository URL. Project will be synchronized with this repository. Supported protocols: `http`, `ssh`, `git`.
+
+* **branch** *(string)*
+  Repository branch.
+
+* **name** *(string)*
+  Project name.
+
+* **authenticationType** *(string)*
+  Authentication type that will be used to get access to the repository. Allowed values: `password`, `ssh_key`.
+
+* **username** *(string)*
+  Username for authentication to the repository.
+
+* **password** *(string)*
+  Password for authentication to the repository.
+
+* **sshPrivateKey** *(string)*
+  SSH private key for authentication to the repository.
```

Comment:
* required if authenticationType is `ssh_key` ?

Reply:
Actually, I haven't planned to require an ssh key, because you probably already set up ssh access to the git serve by using the default ssh key. Do you think this should be a required field?

Reply:
The private key is either there because we generated it or because the use provided it. Either way, the private key is required, it's just the difference in us generating it vs user supplying it.